### ISSUE TITLE
Magic command for convenient list of the baseline devices

### ISSUE THREAD

**Issue by mrakitin** [NSLS-II/bluesky#1171]
## Expected Behavior

There should be a magic command (`%sd` or `%bl` or `%baseline`) for a convenient printout of a list of the baseline devices.

## Current Behavior

It seems we don't have one.

## Possible Solution

Implementation can be very similar to the new `labels` feature available via `%wa`.

## Context

@cmazzoli pointed it out that it would be really convenient to have such a feature.

**Comment by tacaswell** [comment]
`sd.baseline`?

**Comment by danielballan** [comment]
I don't think adding a new magic like `%baseline` would be meaningfully more convenient than the existing `sd.baseline`. This seems like a knowledge-sharing issue. OK to close?

**Comment by mrakitin** [comment]
Not sure it will look as nice as a list of positioners returned by `%wa`, like here: https://github.com/NSLS-II-CSX/xf23id1_profiles/pull/16#issuecomment-485028505. Namely, the limits as well as current values can be displayed without adding `labels` to the existing `sd.baseline` devices.

**Comment by danielballan** [comment]
I see. That prompts me to wonder if the list of things that `%wa` displays should be the same as the list of things that `sd.baseline` records. Perhaps a "bluesky application" is emerging here, which is distinct from what we would want in a _library_.

**Comment by prjemian** [comment]
No.  Including all of %wa content in sd.baseline should not be a default.
That goes beyond the scope of how to make a report of baseline devices.

**Comment by awalter-bnl** [comment]
I tend to agree with @prjemian on this. I think reporting of the baseline devices and printing to the screen a useful list are 2 distinct items. IMO we should be doing both, probably working from the same 'list' of 'baseline devices' but as two seperate processes that work with this list. 

However looking at the example from the link above we also need to be somehow able to 'group' and 'subgroup' the various devices (they have 'groups' like tardis, optics, source and motors with each subgrouped into 'motors' and 'signals'. So the question is should we provides this 'grouping' in the list of baseline devices or elsewhere? and what mechanism do we want to use. 

Personally if this the logical grouping that the beamline thinks about for the devices they use then the grouping needs to be done somewhere in the ophyd layer, so that things like 'group.subgroup.device' can work so that users can tab to complete down the 'logical' levels in order to find the device that they can't remember the name of. 

I am guessing we should also add some functionality that formats the line for each 'device' being printed which can be assigned (probably at a group or sub-group level). for eg. in the example from the link the subgroups 'motors' and 'signals' would have a (potentially customizable) formatting option associated with them that displays the required information for each device in that subgroup.
